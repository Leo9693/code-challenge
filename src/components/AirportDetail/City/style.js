import styled, { keyframes } from "styled-components";
import { fadeInRight } from "react-animations";

export const AnimatedCity = styled.div`
    animation: ${keyframes`${fadeInRight}`} 1.3s;
`;
