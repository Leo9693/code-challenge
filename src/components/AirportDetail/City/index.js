import React from "react";
import { AnimatedP } from "../style";
import { AnimatedCity } from "./style";
function index(props) {
    const { cityCode, cityName, timeZoneName } = props.city;
    return (
        <AnimatedCity>
            <h3>City:</h3>
            <AnimatedP>CityCode: {cityCode}</AnimatedP>
            <AnimatedP>CityName: {cityName}</AnimatedP>
            <AnimatedP>TimeZoneName: {timeZoneName}</AnimatedP>
        </AnimatedCity>
    );
}

export default index;
