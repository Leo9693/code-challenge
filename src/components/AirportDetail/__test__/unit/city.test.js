import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import City from "../../City";
Enzyme.configure({ adapter: new Adapter() });

test("test city component", () => {
    const wrapper = shallow(
        <City
            city={{
                cityCode: "",
                cityName: "",
                timeZoneName: ""
            }}
        />
    );
    expect(wrapper.debug()).toMatchSnapshot();
});
