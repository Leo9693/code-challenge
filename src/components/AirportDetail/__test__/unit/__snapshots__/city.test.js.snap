// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test city component 1`] = `
"<styled.div>
  <h3>
    City:
  </h3>
  <styled.p>
    CityCode: 
  </styled.p>
  <styled.p>
    CityName: 
  </styled.p>
  <styled.p>
    TimeZoneName: 
  </styled.p>
</styled.div>"
`;
